simphony/testing/abc_check_engine.py

```diff
@@ -35,173 +39,186 @@ def create_dataset_items(self):
+        return method(*args, **kwrds)
 
     def test_get_missing_dataset(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
-            with self.assertRaises(ValueError):
-                handle.get_dataset('foo')
+        handle = self.engine_factory()
+        with self.assertRaises(ValueError):
+            handle.get_dataset('foo')
 
     def test_add_dataset_empty(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
-            handle.add_dataset(self.create_dataset(name='test'))
-            ds = handle.get_dataset("test")
+        handle = self.engine_factory()
+        reference = self.create_dataset(name='test')
+        handle.add_dataset(reference)
+        ds = handle.get_dataset("test")
 
-            self.assertEqual("test", ds.name)
-            self.check_emtpy_dataset(ds)
+        self.compare_operation(reference, ds, testcase=self)
 
     def test_add_dataset_empty_data(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
-            handle.add_dataset(self.create_dataset(name='test'))
-            pc = handle.get_dataset("test")
-            self.assertEqual(DataContainer(), pc.data)
-            self.assertEqual(0, len(pc.data))
+        handle = self.engine_factory()
+        handle.add_dataset(self.create_dataset(name='test'))
+        ds = handle.get_dataset("test")
+        self.assertEqual(DataContainer(), ds.data)
+        self.assertEqual(0, len(ds.data))
 
     def test_add_dataset_data_copy(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
-            handle.add_dataset(self.create_dataset(name='test'))
-            pc = handle.get_dataset("test")
-            data = pc.data
-            data[CUBA.NAME] = 'somename'
-
-            # Since the returned data is always a copy,
-            #  therefore the pc.data should not have changed
-            self.assertNotIn(CUBA.NAME, pc.data)
-            # And the length should be still zero
-            self.assertEqual(0, len(pc.data))
-            pc.data = data
-            # This time we replaced the pc.data,
-            #  therefore it should have been changed
-            self.assertIn(CUBA.NAME, pc.data)
-            # The length also should have been changed
-            self.assertEqual(1, len(pc.data))
+        handle = self.engine_factory()
+        handle.add_dataset(self.create_dataset(name='test'))
+        ds = handle.get_dataset("test")
+        data = ds.data
+        data[CUBA.NAME] = 'somename'
+
+        # Since the returned data is always a copy,
+        #  therefore the ds.data should not have changed
+        self.assertNotIn(CUBA.NAME, ds.data)
+        # And the length should be still zero
+        self.assertEqual(0, len(ds.data))
```

Comment:
This test has to be reworked.  It is based on the assumption that the dataset's DataContainer `data` is empty.  

In general, it looks like goal of this test was to check that the getting and setting of `data` of the DataSets  takes place through a copy (#84)